Update core tools instructions

anthonychu opened the pull request:
@ankitkumarr Please review for Core Tools install. Removed .NET Core requirement and updated command.

@pragnagopa Please review custom handlers instructions.

@takyyon We don't have VS Code support for custom handlers yet, we can use the CLI instructions.

ammanifold commented:
Thanks @anthonychu ! I will review today.

@btardif @takyyon can you both please take a look also?

ammanifold commented on server/src/quickstart/coretoolsDirectPublish-custom-v2.md:
```diff
@@ -0,0 +1,47 @@
+### Install dependencies
+
+Before you can get started, you should <a href="https://go.microsoft.com/fwlink/?linkid=2016195" target="_blank">install Node.js</a> which includes npm, which is how you will obtain the Azure Functions Core Tools. If you prefer not to install Node.js, see the other installation options in our <a href="https://go.microsoft.com/fwlink/?linkid=2016192" target="_blank">Core Tools reference</a>.
```
Suggestion:
Before you can get started, you should <a href="https://go.microsoft.com/fwlink/?linkid=2016195" target="_blank">install Node.js</a> which includes npm. This is how you will obtain the Azure Functions Core Tools.

anthonychu replied:
That sounds much better. Will update. 🙂

ammanifold commented on server/src/quickstart/coretoolsDirectPublish-v2.md:
```diff
@@ -1,10 +1,10 @@
 ### Install dependencies
 
-Before you can get started, you should <a href="https://go.microsoft.com/fwlink/?linkid=2016373" target="_blank">install .NET Core 2.1</a>. You should also <a href="https://go.microsoft.com/fwlink/?linkid=2016195" target="_blank">install Node.JS</a> which includes npm, which is how you will obtain the Azure Functions Core Tools. If you prefer not to install Node, see the other installation option ...
+Before you can get started, you should <a href="https://go.microsoft.com/fwlink/?linkid=2016195" target="_blank">install Node.js</a> which includes npm, which is how you will obtain the Azure Functions Core Tools. If you prefer not to install Node.js, see the other installation options in our <a href="https://go.microsoft.com/fwlink/?linkid=2016192" target="_blank">Core Tools reference</a>.
```
Same update here, I think there are too many "which" in first sentence :)

pragnagopa commented on server/src/quickstart/vscodeDirectPublish-v2.md:
```diff
@@ -4,9 +4,7 @@ Before you can get started, you should <a href="https://go.microsoft.com/fwlink/
-<MarkdownHighlighter>npm install -g azure-functions-core-tools</MarkdownHighlighter>
-
-The Core Tools make use of <a href="https://go.microsoft.com/fwlink/?linkid=2016373" target="_blank">.NET Core 2.1</a>, so you should install that, too.
+<MarkdownHighlighter>npm install -g azure-functions-core-tools@3 --unsafe-perm true</MarkdownHighlighter>
```
rename the files to vsCodeDirectPublish-v3?

anthonychu replied:
I _think_ `-v2` means v2 of the portal.

takyyon replied:
Actually let's make a copy of all the v2 files and rename them to v3. And these changes can then go to v3.

ammanifold replied:
Or we could do -localDevExperience (or something of the like). I think we should be avoiding v2/v3 in functions land (unless addressing the runtime) to minimize confusion

takyyon replied:
Yeah, let's do something like: `{filename}-react-localDevExperience`

anthonychu replied:
So is this a rename or a copy?

takyyon replied:
Copy!

pragnagopa commented on server/src/quickstart/coretoolsDirectPublish-v2.md:
```diff
@@ -1,10 +1,10 @@
 Run the following command to install the Core Tools package:
 
-<MarkdownHighlighter>npm install -g azure-functions-core-tools</MarkdownHighlighter>
+<MarkdownHighlighter>npm install -g azure-functions-core-tools@3 --unsafe-perm true</MarkdownHighlighter>
```
Same here. Rename the file?

pragnagopa commented on server/src/quickstart/coretoolsDirectPublish-custom-v2.md:
```diff
@@ -0,0 +1,47 @@
+
+<MarkdownHighlighter>func azure functionapp publish {functionAppName} --no-build --force</MarkdownHighlighter>
+
+You may be prompted to sign into Azure. Follow the onscreen instructions. The actual command may depend on the language used to build your custom handler.
```
Is this needed: `The actual command may depend on the language used to build your custom handler` ? As we will not be adding any language specific commands for custom handlers?

anthonychu replied:
Probably not. I'll remove. @pragnagopa  Do you know when Core Tools will be updated to publish custom handlers and what the command will look like (does it need the flags)?

pragnagopa replied:
@yojagad - Will changes be part of the next core tools release?

yojagad replied:
I just tested publish with the changes that already went in. I was able to just do func azure functionapp publish {functionAppName} without the --no-build and --force flags and verified that the trigger worked.

takyyon commented:
@anthonychu Is the `coretoolsDirectPublish-custom-v2.md` entirely different from `coretoolsDirectPublish-v2.md` or just some minor changes? In case of latter, I am thinking if we can introduce a `<CustomStack>` to manage that instead of a separate file.

anthonychu commented:
> @anthonychu Is the `coretoolsDirectPublish-custom-v2.md` entirely different from `coretoolsDirectPublish-v2.md` or just some minor changes? In case of latter, I am thinking if we can introduce a `<CustomStack>` to manage that instead of a separate file.

Yep after addressing PR comments, the only thing that changed was the addition of 1 sentence: https://github.com/Azure/azure-functions-ux/blame/0ec96f17e1997f962ea5f7f4192fbec15e5c21a6/server/src/quickstart/coretoolsDirectPublish-custom-v2.md#L27

takyyon commented:
> > @anthonychu Is the `coretoolsDirectPublish-custom-v2.md` entirely different from `coretoolsDirectPublish-v2.md` or just some minor changes? In case of latter, I am thinking if we can introduce a `<CustomStack>` to manage that instead of a separate file.
> 
> Yep after addressing PR comments, the only thing that changed was the addition of 1 sentence: https://github.com/Azure/azure-functions-ux/blame/0ec96f17e1997f962ea5f7f4192fbec15e5c21a6/server/src/quickstart/coretoolsDirectPublish-custom-v2.md#L27

Gotcha! In that case we can do something like this,

`<StackInstructions>Instructions here</StackInstructions>`
`<StackInstructions customStack={true}>Custom Instructions here</StackInstructions>`

anthonychu commented:
@takyyon Added `<StackInstructions>`. Is this a new component that you'll be building to hide/show sections based on stack? If yes, wonder if it should be something like this so it's more flexible:

```
<StackInstructions includeStacks={['custom']}>...</StackInstructions>
```

takyyon commented:
> @takyyon Added `<StackInstructions>`. Is this a new component that you'll be building to hide/show sections based on stack? If yes, wonder if it should be something like this so it's more flexible:
> 
> ```
> <StackInstructions includeStacks={['custom']}>...</StackInstructions>
> ```

Yeah, but I am not really sure how much support is there for component props from the library. Let's go with what just pushed in. I'll play around with it once it gets into dev and update accordingly. Also, would you please mind creating a copy of the file with the name from the other comment.
